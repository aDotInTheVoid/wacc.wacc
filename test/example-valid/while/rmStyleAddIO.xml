<unit>
  <main>
    <addr_local>0</addr_local>
    <num>0</num>
    <assign_do/>
    <addr_local>1</addr_local>
    <num>0</num>
    <assign_do/>
    <str>"Enter the first number: "</str>
    <print><kind>str</kind><newline>false</newline></print>
    <addr_local>0</addr_local>
    <read><ty>int</ty></read>
    <str>"Enter the second number: "</str>
    <print><kind>str</kind><newline>false</newline></print>
    <addr_local>1</addr_local>
    <read><ty>int</ty></read>
    <str>"Initial value of x: "</str>
    <print><kind>str</kind><newline>false</newline></print>
    <local>0</local>
    <print><kind>i32</kind><newline>true</newline></print>
    <while>
      <cond><local>1</local><num>0</num><op>gt</op></cond>
      <body>
        <str>"(+)"</str>
        <print><kind>str</kind><newline>false</newline></print>
        <addr_local>0</addr_local>
        <local>0</local>
        <num>1</num>
        <op>add</op>
        <assign_do/>
        <addr_local>1</addr_local>
        <local>1</local>
        <num>1</num>
        <op>sub</op>
        <assign_do/>
      </body>
    </while>
    <str>""</str>
    <print><kind>str</kind><newline>true</newline></print>
    <str>"final value of x: "</str>
    <print><kind>str</kind><newline>false</newline></print>
    <local>0</local>
    <print><kind>i32</kind><newline>true</newline></print>
  </main>
</unit>