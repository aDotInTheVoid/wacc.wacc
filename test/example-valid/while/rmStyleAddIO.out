Enter the first number: Enter the second number: Initial value of x: 12
(+)(+)(+)(+)(+)(+)(+)(+)(+)(+)(+)(+)(+)(+)
final value of x: 26
